### PR TITLE
[prometheus-cloudwatch-exporter] - Added option to enable sts regional endpoints

### PR DESCRIPTION
**Pull request by lewisstrong1991**
#### What this PR does / why we need it:

This PR fixes the issue that when using STS to assume a role the global endpoint is used by default. If this is not accessible and a regional endpoint is configured the launched pods/containers will get in a crash loop. By setting the ENV variable `AWS_STS_REGIONAL_ENDPOINTS=regional` this tells the SDK to use the regional endpoint instead of the global endpoint.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by lewisstrong1991** [comment]
@asherf @gianrubio @torstenwalter - Any chance someone can please take a look at this?

File: charts/prometheus-cloudwatch-exporter/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "0.8.0"
 description: A Helm chart for prometheus cloudwatch-exporter
 name: prometheus-cloudwatch-exporter
-version: 0.12.1
+version: 0.12.2
```
Comment (torstenwalter) [inline charts/prometheus-cloudwatch-exporter/Chart.yaml:5]: Can you change it to 0.13.0. As it's a new feature it's worth incrementing the minor version.

**Review by torstenwalter — CHANGES_REQUESTED** [review]
LGTM - just a small change. Just ping we afterwards and I'll merge it.

File: charts/prometheus-cloudwatch-exporter/Chart.yaml (hunk 1 above)
Comment (lewisstrong1991) [reply charts/prometheus-cloudwatch-exporter/Chart.yaml]: No worries! Bumped and also fixed the version in the `CHANGELOG.md` as realised i'd mistyped that.

**Comment by lewisstrong1991** [comment]
@torstenwalter - Fixed and all tests passed

**Comment by torstenwalter** [comment]
Don't we have a duplicate `env` section with this in case the endpoint and aws secret are defined?

**Comment by lewisstrong1991** [comment]
They should be mutually exclusive. STS would only be used when specifying an AWS Role and not used where AWS credentials are used. Could look to code this in an IF-ELSE statement?

**Comment by torstenwalter** [comment]
Would make sense if they are exclusive.

**Comment by lewisstrong1991** [comment]
@torstenwalter - Made a change and tested this locally. Will only be applied when a AWS role is set in the values.

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
@lewisstrong1991 thanks for implementing it.

**Pull request merged**